tests/framework/common.go

```diff
@@ -328,13 +328,21 @@ nodeRegistration:
 	return nil
 }
 
+// WithLocalProxyMode sets the LocalProxyMode for OSM
+func WithLocalProxyMode(mode configv1alpha2.LocalProxyMode) InstallOsmOpt {
+	return func(opts *InstallOSMOpts) {
+		opts.LocalProxyMode = mode
+	}
+}
+
```

Comment:
This deviates from how other MeshConfig attributes are configured. Could you follow the same patter, and directly update MeshConfig in tests as:
```
installOpts := Td.GetOSMInstallOpts()
installOpts.LocalProxyMode = ...
Expect(Td.InstallOSM(installOpts)).To(Succeed())
```

Reply:
To do that, I'd have to create a separate `It()` block. My goal was to re-use the existing testing setup for pod_client_server and just add a separate case since they're related (see my comment above). I feel like the option pattern is an unobtrusive approach that maintains existing contracts while allowing further flexibility.